podcasts/StreamingEpisodeArtwork.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import Kingfisher
+
+/// Extracts artwork from a streaming episode (if there's any)
+class StreamingEpisodeArtwork {
+    static let shared = StreamingEpisodeArtwork()
+
+    private lazy var subscribedPodcastsCache: ImageCache = {
+        let path = (NSHomeDirectory() as NSString).appendingPathComponent("Documents/artworkv3")
```

Comment:
This is the same cache used in `ImageManager`. It would be good to extract this to have this logic in a single place.

I haven't done it yet because I'd like to hear what other thinks about this approach.

Reply:
I like this approach since it's a bit more semantic and task oriented than the ImageManager.

I wonder if we can just change the cache property in the ImageManager to `private (set) var subscribedPodcastsCache: ImageCache`, that way we can still access it elsewhere, and don't need to refactor the ImageManager. 

If we do that this class will no longer need to be a singleton since it'll be using the singleton cache property from the ImageManager already. 

Thoughts?

Reply:
I made some changes.

I wasn't able to fully make the `StreamingEpisodeArtwork` not a singleton so open to any ideas on how to improve it.

I could move all the logic to `ImageManager` but, oh well, it's a Singleton too. :|